Add 404 page

drasmuss opened the pull request:
Uses the `sphinx-notfound-page` extension to make this easy. Depends on https://github.com/nengo/nengo-bones/pull/115.

drasmuss committed:
Add 404 page

tbekolay commented on nengo_sphinx_theme/theme/404.html:
```diff
@@ -0,0 +1,22 @@
+          were on doesn't exist in this version.
+        </p>
+        <form action="" method="get">
+        <input type="text" name="q" aria-labelledby="search-documentation" value="" />
```
It would be nice to fill in the search value automatically using the requested page (from the URL that was originally requested -- you can get this in JS with `window.location`).

drasmuss replied:
Done

Commit:
Automatic update to NengoBones 3dd9701

3dd9701 - Display remote pip environment
605b2bb - Run format-notebook on configuration.ipynb
1b78d39 - Apply flake8/pylint checks in format-notebook

drasmuss committed:
Apply bones-format-notebook

drasmuss committed:
Fix pylint 2.6 warnings

hunse commented:
The search box wasn't working. It would send the query to the current (404) page, rather than the search page (e.g. https://www.nengo.ai/nengo/404.html?q=mysearchterm). I switched things to using the code from the search in the sidebar, which works. There's a lot of stuff in there that's just magic to me (and I'm just blindly copy-pasting), so let me know if there's a better way to do it.

drasmuss commented:
LGTM
